tests/unit/models/governing-body-test.js

```diff
@@ -0,0 +1,828 @@
+
+      assert.true(isValid);
+      assert.notOk(model.error, 'model.error should be undefined');
+
+      otherBodiesStub.restore();
+    });
+
+    test("it returns no error when the period of another governing body is strictly after the model's period", async function (assert) {
+      const otherModel = this.store().createRecord('governing-body', {
+        startDate: new Date('2024-01-01'),
+        endDate: new Date('2026-01-01'),
+      });
+
+      const model = this.store().createRecord('governing-body', {
+        startDate: new Date('2020-01-01'),
+        endDate: new Date('2023-01-01'),
+      });
+
+      const otherBodiesStub = sinon.stub(model, 'getOtherTimedGoverningBodies');
+      otherBodiesStub.resolves([otherModel]);
+
+      const isValid = await model.validate();
+
+      assert.true(isValid);
+      assert.notOk(model.error, 'model.error should be undefined');
+
+      otherBodiesStub.restore();
+    });
+
+    test("it returns no error when the period of another governing body is strictly before the model's period", async function (assert) {
+      const otherModel = this.store().createRecord('governing-body', {
+        startDate: new Date('2020-01-01'),
+        endDate: new Date('2023-01-01'),
+      });
+
+      const model = this.store().createRecord('governing-body', {
+        startDate: new Date('2024-01-01'),
+        endDate: new Date('2026-01-01'),
+      });
+
+      const otherBodiesStub = sinon.stub(model, 'getOtherTimedGoverningBodies');
+      otherBodiesStub.resolves([otherModel]);
+
+      const isValid = await model.validate();
+
+      assert.true(isValid);
+      assert.notOk(model.error, 'model.error should be undefined');
+
+      otherBodiesStub.restore();
+    });
+
+    test("it returns two errors when the start date of another governing body is strictly between the model's start and end date", async function (assert) {
+      const otherModel = this.store().createRecord('governing-body', {
+        startDate: new Date('2024-06-06'),
+        endDate: new Date('2026-01-01'),
+      });
+
+      const model = this.store().createRecord('governing-body', {
+        startDate: new Date('2024-01-01'),
+        endDate: new Date('2025-01-01'),
+      });
+
+      const otherBodiesStub = sinon.stub(model, 'getOtherTimedGoverningBodies');
+      otherBodiesStub.resolves([otherModel]);
+
+      const isValid = await model.validate();
+
+      assert.false(isValid);
+      assert.ok(model.error, 'model.error should not be undefined');
+      assert.strictEqual(Object.keys(model.error).length, 2);
+      assert.propContains(model.error, {
+        startDate: {
+          message: 'Geen overlap',
+        },
+        endDate: {
+          message: 'Geen overlap',
+        },
+      });
+
+      otherBodiesStub.restore();
+    });
+
+    test("it returns no error when the model's period is strictly within the period of another governing body", async function (assert) {
```

Comment:
**question:** Is it the expected behavior? It isn't overlapping?

Reply:
Not sure whether it is the expected behaviour, but that is how it worked before migrating to the Joi validation. So I kept it for this refactoring, although I agree that this situation is overlapping. It is probably a good idea to have a follow up to look at what "overlap" precisely means in the context of governing bodies and change our implementation accordingly.

Reply:
Can you add a Technical story in the backlog to not forget?